### ISSUE TITLE
HPy: New C-API for Python

### ISSUE THREAD

**Issue by fniephaus** [oracle/graalpython#285]
### TL;DR

We plan to design and build a new C-API for Python that makes it easier for Python developers to interact with native code and ship native code with their packages.

### Goals

As contributors to the [HPy project](https://hpyproject.org/), we are actively participating in the design of a new C-API for Python. We also plan to implement this API for GraalPy. For more information and updates on this effort, please visit the [HPy website](https://hpyproject.org/) or check out our HPy blog posts on the [GraalVM blog](https://medium.com/graalvm).

### Non-Goal

Drop support for the legacy C-API in CPython.
